Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1

```diff
@@ -1,175 +1,178 @@
+        Write-Verbose "Completed Register-Environment cmdlet call for environment : $environmentName"
 
-        Write-Output (Get-LocalizedString -Key "Deployment status for machine '{0}' : '{1}'" -ArgumentList $displayName, $status)
+        Write-Verbose "Starting Get-EnvironmentResources cmdlet call on environment name: $environmentName"
+        $resources = Get-EnvironmentResources -Environment $environment
 
-        if ($status -ne "Passed")
+        if ($resources.Count -eq 0)
         {
-            Write-Telemetry "DTLSDK_Error" $deploymentResponse.DeploymentSummary
-            Write-Verbose $deploymentResponse.Error.ToString()
-            $errorMessage =  $deploymentResponse.Error.Message
-            throw $errorMessage
+            Write-Telemetry "Input_Validation" "No machine exists for given environment"
+            throw (Get-VstsLocString -Key "PS_TM_NoMachineExistsUnderEnvironment0ForDeployment" -ArgumentList $environmentName)
         }
-    }
-}
-else
-{
-    [hashtable]$Jobs = @{} 
-    $dtlsdkErrors = @()
 
-    foreach($resource in $resources)
+        $resourcesPropertyBag = Get-ResourcesProperties -resources $resources
+    }
+    catch
     {
-        $resourceProperties = $resourcesPropertyBag.Item($resource.Id)
-        $machine = $resourceProperties.fqdn
-        $displayName = $resourceProperties.displayName
-        Write-Output (Get-LocalizedString -Key "Deployment started for machine: '{0}'" -ArgumentList $displayName)
+        Write-Telemetry "Task_InternalError" $_.exception.Message
+
+        throw
+    }
 
-        $job = Start-Job -ScriptBlock $RunPowershellJob -ArgumentList $machine, $scriptPath, $resourceProperties.winrmPort, $scriptArguments, $initializationScriptPath, $resourceProperties.credential, $resourceProperties.protocolOption, $resourceProperties.skipCACheckOption, $enableDetailedLoggingString, $sessionVariables
-        $Jobs.Add($job.Id, $resourceProperties)
+    if($runPowershellInParallel -eq "false" -or  ( $resources.Count -eq 1 ) )
+    {
+        foreach($resource in $resources)
+        {
+            $resourceProperties = $resourcesPropertyBag.Item($resource.Id)
+            $machine = $resourceProperties.fqdn
+            $displayName = $resourceProperties.displayName
+            Write-Output (Get-VstsLocString -Key "PS_TM_DeploymentStartedForMachine0" -ArgumentList $displayName)
+
+            $deploymentResponse = Invoke-Command -ScriptBlock $RunPowershellJob -ArgumentList $machine, $scriptPath, $resourceProperties.winrmPort, $scriptArguments, $initializationScriptPath, $resourceProperties.credential, $resourceProperties.protocolOption, $resourceProperties.skipCACheckOption, $enableDetailedLoggingString, $sessionVariables
```

Comment:
ensure you are not doing any import from agent set-up in job.ps1. All the required imports should be form $psroot or task folder only